File: 3Sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        n=len(nums)
        A=[]
        nums.sort()
        for i in range(n-1):
            d=dict()
            if (i>0 and nums[i]==nums[i-1]): continue
            if (nums[i]>0):break
            for j in range(i+1,n):
                if -(nums[i]+nums[j]) in d:
                    temp=[-(nums[i]+nums[j]),nums[i],nums[j]]
                    if (len(A)>0 and A[-1]==temp): continue
                    A.append(temp)
                d[nums[j]]=j
        return A
__import__("atexit").register(lambda: open("display_runtime.txt","w").write("000"))

File: 3Sum.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var threeSum = function(nums) {
    let n = nums.length;
    let A = [];
    nums.sort((a, b) => a - b);
    for (let i = 0; i < n-2; i++) {
        if (i > 0 && nums[i] === nums[i - 1]) continue;
        let j=i+1
        let k=n-1
        while(j<k){
            let sum=(nums[i]+nums[j]+nums[k])
            if (sum<0) j++
            else if (sum>0) k--
            else {
                A.push([nums[i],nums[j],nums[k]])
                k--
                j++
                while(j<k && nums[j]===nums[j-1]) j++
                while(j<k && nums[k]===nums[k+1]) k--
            }
        }
    }
    return A
};

